pkg/controller/volume/persistentvolume/pv_controller_test.go

```diff
@@ -461,3 +467,110 @@ func TestDelayBindingMode(t *testing.T) {
 		}
 	}
 }
+
+func TestAnnealMigrationAnnotations(t *testing.T) {
+	defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.CSIMigration, true)()
+
+	const testPlugin = "non-migrated-plugin"
+	const gcePlugin = "kubernetes.io/gce-pd"
+	const gceDriver = "pd.csi.storage.gke.io"
+	tests := []struct {
+		name                 string
+		volumeAnnotations    map[string]string
+		expVolumeAnnotations map[string]string
+		claimAnnotations     map[string]string
+		expClaimAnnotations  map[string]string
+		migratedDriverGates  []featuregate.Feature
+	}{
+		{
+			name:                 "migration on for GCE",
+			volumeAnnotations:    map[string]string{pvutil.AnnDynamicallyProvisioned: gcePlugin},
+			expVolumeAnnotations: map[string]string{pvutil.AnnDynamicallyProvisioned: gcePlugin, pvutil.AnnMigratedTo: gceDriver},
+			claimAnnotations:     map[string]string{pvutil.AnnStorageProvisioner: gcePlugin},
+			expClaimAnnotations:  map[string]string{pvutil.AnnStorageProvisioner: gcePlugin, pvutil.AnnMigratedTo: gceDriver},
+			migratedDriverGates:  []featuregate.Feature{features.CSIMigrationGCE},
+		},
+		{
+			name:                 "migration off for GCE",
+			volumeAnnotations:    map[string]string{pvutil.AnnDynamicallyProvisioned: gcePlugin},
+			expVolumeAnnotations: map[string]string{pvutil.AnnDynamicallyProvisioned: gcePlugin},
+			claimAnnotations:     map[string]string{pvutil.AnnStorageProvisioner: gcePlugin},
+			expClaimAnnotations:  map[string]string{pvutil.AnnStorageProvisioner: gcePlugin},
+			migratedDriverGates:  []featuregate.Feature{},
```

Comment:
Will this test fail when GCE migration is on by default, and should we dump that problem on future us?

I vote yes.